tests/test_runner.py

```diff
@@ -65,6 +65,24 @@
             {},
             id="no-type-dict",
         ),
+        *(
+            [
+                pytest.param(
+                    "tests.script_runner.parameter_inputs:str_or_int_parameter",
+                    [{"name": "my_str_or_int", "value": "hi there"}],
+                    "type given: str",
+                    id="str-or-int-given-str",
+                ),
+                pytest.param(
+                    "tests.script_runner.parameter_inputs:str_or_int_parameter",
+                    [{"name": "my_str_or_int", "value": "3"}],
+                    "type given: int",
+                    id="str-or-int-given-int",
+                ),
+            ]
+            if _PYDANTIC_VERSION > 1
```

Comment:
Why not use pytest.skipif with pytest.params?

Reply:
Yep, that would have made sense. No longer needed though as I reordered the union elements so the test passes.